src/similarity.py

```diff
@@ -223,11 +225,36 @@ def _cosine_fast(
         row_ind, col_ind = scipy.optimize.linear_sum_assignment(
             cost_matrix, maximize=True
         )
+    matches = 0
     score = 0.0
+    total_intensity = 0.0
+    matched_intensity = 0.0
+    max_contribution = 0.0
+    # signals with contribution to cosine score greater 5%
+    n_greq_2p = 0
+
     row_mask = np.zeros_like(row_ind, np.bool_)
     col_mask = np.zeros_like(col_ind, np.bool_)
     for (i, row), (j, col) in zip(enumerate(row_ind), enumerate(col_ind)):
-        if cost_matrix[row, col] > 0.0:
-            score += cost_matrix[row, col]
+        total_intensity += spec.intensity[row] + spec_other.intensity[col]
+        pair_score = cost_matrix[row, col]
+        if pair_score > 0.0:
+            score += pair_score
+            matched_intensity += spec.intensity[row] + spec_other.intensity[col]
             row_mask[i] = col_mask[j] = True
-    return score, (row_ind[row_mask], col_ind[col_mask])
+            matches += 1
+            if pair_score >= 0.02:
+                n_greq_2p += 1
+            if pair_score > max_contribution:
+                max_contribution = pair_score
+
+    if total_intensity > 0:
```

Comment:
Normalizing by total intensity is not necessary because the intensity vectors are converted to unit vectors as part of the similarity calculations.